jsrc/j.c

```diff
@@ -58,6 +58,9 @@ I Bnum[22][8] =
     CBAIVAL(B01, 0),  CBAIVAL(B01, 1),  CBAIVAL(INT, 2),   CBAIVAL(INT, 3),  CBAIVAL(INT, 4),  CBAIVAL(INT, 5),
     CBAIVAL(INT, 6),  CBAIVAL(INT, 7),  CBAIVAL(INT, 8),   CBAIVAL(INT, 9)};
 
+I Bbool[2][8] = { // The booleans
```

Comment:
For the booleans, instead of array of two, the values could just be introduced individually, removing the need for access macros
```c
AD const jfalse = CBAIVAL(B01, 0);
AD const jtrue  = CBAIVAL(B01, 1);
```

Reply:
All users of the code currently depend on it being a pointer/`A`, the first solution would need all users of it to change to `&jtrue` and `&jfalse`, which is kind of a deal breaker.

Reply:
That is true, those should probably be pointers for now, to keep the semantics consistent.

In the long run, I think it would be better to make use of pointers explicit, not hiding those in typedefs.

Reply:
I've pushed an update with the second suggestion and a todo marker for the `CBAIVAL` macro. Updating any more of this file starts to feel like this is becoming a kitchen sink of changes, making it very hard to review the change and making it very sensitive to merge conflicts if any other updates are pushed.

Reply:
Looks like a great start. Will merge & look forward to follow up PRs.